test/testtokenize.cpp

```diff
@@ -1227,16 +1233,123 @@ class TestTokenizer : public TestFixture {
         ASSERT_THROW(tokenizeAndStringify(code), InternalError);
     }
 
-    void ifAddBraces21() { // #5332 - if (x) label: {} ...
+    void ifAddBracesLabels() {
+        // Labels before statement
+        ASSERT_EQUALS("int f ( int x ) {\n"
+                      "if ( x ) {\n"
+                      "l1 : ; l2 : ; return x ; }\n"
+                      "}",
+                      tokenizeAndStringify("int f(int x) {\n"
+                                           "  if (x)\n"
+                                           "  l1: l2: return x;\n"
+                                           "}"));
+
+        // Labels before {
+        ASSERT_EQUALS("int f ( int x ) {\n"
+                      "if ( x )\n"
+                      "{ l1 : ; l2 : ; return x ; }\n"
+                      "}",
+                      tokenizeAndStringify("int f(int x) {\n"
+                                           "  if (x)\n"
+                                           "  l1: l2: { return x; }\n"
+                                           "}"));
+
+        // Labels before try/catch
+        ASSERT_EQUALS("int f ( int x ) {\n"
+                      "if ( x ) {\n"
+                      "l1 : ; l2 : ;\n"
+                      "try { throw 1 ; }\n"
+                      "catch ( ... ) { return x ; } }\n"
+                      "}",
+                      tokenizeAndStringify("int f(int x) {\n"
+                                           "  if (x)\n"
+                                           "  l1: l2:\n"
+                                           "    try { throw 1; }\n"
+                                           "    catch(...) { return x; }\n"
+                                           "}"));
+
+        // #5332 - if (x) label: {} ...
         const char code[] = "void f() { if(x) label: {} a=1; }";
-        ASSERT_EQUALS("void f ( ) { if ( x ) { label : ; { } } a = 1 ; }", tokenizeAndStringify(code));
+        ASSERT_EQUALS("void f ( ) { if ( x ) { label : ; } a = 1 ; }", tokenizeAndStringify(code));
```

Comment:
This test is kind of redundant now, should I remove it? Or it is better to keep it as a regression test to the ticket?

Reply:
if it doesn't add anything to the testing feel free to remove it.

Reply:
OK, I have removed the old test.